### ISSUE TITLE
Project using Quartz and IdentityServer module breaks in 3.0.1

### ISSUE THREAD

**Issue by haivp3010** [abpframework/abp#4609]
**TL;DR**:
`Volo.Abp.BackgroundWorkers.Quartz` 3.0.1 causes `TokenCleanupBackgroundWorker` in ABP IdentityServer module to be failed to schedule.
The workaround is to increase the default 250-character limit in table `QRTZ_JOB_DETAILS`.

**ABP Framework version**: 3.0.1
**Exception message**:

> Volo.Abp.AbpInitializationException: An error occurred during the initialize Volo.Abp.Modularity.OnApplicationInitializationModuleLifecycleContributor phase of the module Volo.Abp.IdentityServer.AbpIdentityServerDomainModule, Volo.Abp.IdentityServer.Domain, Version=3.0.1.0, Culture=neutral, PublicKeyToken=null: Couldn't store job: Data too long for column 'JOB_CLASS_NAME' at row 1. See the inner exception for details.
 ---> Quartz.JobPersistenceException: Couldn't store job: Data too long for column 'JOB_CLASS_NAME' at row 1
 ---> MySql.Data.MySqlClient.MySqlException (0x80004005): Data too long for column 'JOB_CLASS_NAME' at row 1
   at MySql.Data.MySqlClient.MySqlStream.ReadPacket()
   at MySql.Data.MySqlClient.NativeDriver.GetResult(Int32& affectedRow, Int64& insertedId)
   at MySql.Data.MySqlClient.Driver.GetResult(Int32 statementId, Int32& affectedRows, Int64& insertedId)
   at MySql.Data.MySqlClient.Driver.NextResult(Int32 statementId, Boolean force)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResult()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader(CommandBehavior behavior)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at System.Data.Common.DbCommand.ExecuteNonQueryAsync(CancellationToken cancellationToken)

I did some debugging and it seems to be related to the introduction of the new generic [QuartzPeriodicBackgroundWorkerAdapter](https://github.com/abpframework/abp/blob/3.0.1/framework/src/Volo.Abp.BackgroundWorkers.Quartz/Volo/Abp/BackgroundWorkers/Quartz/QuartzBackgroundWorkerAdapter.cs)

The Quartz package is getting a `JOB_CLASS_NAME` by the following code:

https://github.com/quartznet/quartznet/blob/aede51050c3ca73ce506e05632f22e1aeecd65ec/src/Quartz/Impl/AdoJobStore/StdAdoDelegate.cs#L717
```c#
protected virtual string GetStorableJobTypeName(Type jobType)
{
    if (jobType.AssemblyQualifiedName == null)
    {
        throw new ArgumentException("Cannot determine job type name when type's AssemblyQualifiedName is null");
    }

    return jobType.AssemblyQualifiedNameWithoutVersion();
}
```

which eventually use `Type.FullName` on the generic type:
https://github.com/quartznet/quartznet/blob/28cb7eb9d9cf7bb73e07f1770fc629c58e966ab9/src/Quartz/Util/ObjectExtensions.cs#L14
```c#
public static string AssemblyQualifiedNameWithoutVersion(this Type type)
    => assemblyQualifiedNameCache.GetOrAdd(type, x => x.FullName + ", " + x.GetTypeInfo().Assembly.GetName().Name);
```

And that would result in a `JOB_CLASS_NAME` like this, exceeding the default 250-character limit in Quartz tables creation SQL script
```
Volo.Abp.BackgroundWorkers.Quartz.QuartzPeriodicBackgroundWorkerAdapter`1[[Volo.Abp.IdentityServer.Tokens.TokenCleanupBackgroundWorker, Volo.Abp.IdentityServer.Domain, Version=3.0.1.0, Culture=neutral, PublicKeyToken=null]], Volo.Abp.BackgroundWorkers.Quartz
```

My current workaround is to update my database table to increase the limit. But it's still worth mentioning here since someone running the same setup may face the issue.

**Comment by liangshiw** [comment]
You are right.
You can fix this by changing the table field length. Thanks.

**Comment by olicooper** [comment]
I have just run in to this issue too. Is there a reason we need to use `Type.FullName` (fully qualified name) instead of the shorter `Type.Name`?

**Comment by liangshiw** [comment]
Hi @olicooper ,

You can change the quartz source code, because it use `FullName`.
Yes, We can change the default length in https://github.com/abpframework/abp/issues/3433.
